Model/CreditmemoByOrderIncrementId.php

```diff
@@ -100,6 +102,63 @@ public function filterItemsToBeRefunded(Order $order, CreditmemoInterface $credi
         return $selectedItemsToRefund;
     }
 
+    /**
+     * @param Order $order
+     * @param CreditmemoInterface $creditmemo
+     * @param CreditmemoInterface $newCreditmemo
+     */
+    protected function addBackToStockStatus(Order $order, CreditmemoInterface $creditmemo, CreditmemoInterface $newCreditmemo)
+    {
+        $itemsToBackToStock = [];
+        /** @var \Magento\Sales\Model\Order\Item $orderItem */
+        foreach ($order->getAllItems() as $orderItem){
+            /** @var \Magento\Sales\Model\Order\Creditmemo\Item $inputItem */
+            foreach ($creditmemo->getItems() as $inputItem){
+                $itemsToBackToStock = $this->getBackToStockItems(
```

Comment:
Hey @ahasAmp, maybe this should be a function that returns a boolean instead? That way we're not passing an array back and forth unnecessarily. For example:
```php
<?php
$itemsToBackToStock = [];

foreach ($creditMemo->getItems() as $creditmemoItem) {
    if (!$this->shouldGoBackToStock($creditmemoItem, $orderItem)) {
        continue;
    }

    $itemsToBackToStock[] = $creditmemoItem->getSku();
}

// Remove dupes
$itemsToBackToStock = array_unique($itemsToBackToStock);
```
And then the `shouldGoBackToStock` function could do something like:
```php
<?php

return $orderItem->getSku() === $creditmemoItem->getSku()
    && $creditmemoItem->getExtensionAttributes()->getBackToStock();
```
It might be worth checking whether the creditmemo item has extension attributes before hand however otherwise you could be calling `getBackToStock` on a `null`